File: webapp/src/services/OfficeService.ts
import Office from "@/models/Office";
import restClient from "./RestClient";

export class OfficeService {
    public getOffices(streetName: string): Promise<Office[]> {
        return restClient.get<Office[]>("offices", { streetName: streetName });
    }
}

let officeService = new OfficeService()

export default officeService

File: server/App.API/App.Domain/Domain/Services/OfficeService.cs
using App.Context.Domain.Models;
using App.Context.Domain.Repository;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace App.Context.Domain.Services
{
    public class OfficeService : IOfficeService
    {
        private readonly IOfficeRepository _iOfficeRepository;
        public OfficeService(IOfficeRepository officeRepository) => _iOfficeRepository = officeRepository;

        public async Task<IEnumerable<Office>> GetOffices()
            => await _iOfficeRepository.GetAllOffices();

        public async Task<IEnumerable<Office>> GetOfficesByStreetName(string streetName)
            => await _iOfficeRepository.GetOfficesByStreetName(streetName);
    }
}
